shopinvader_api_address/models/res_partner.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from fastapi import status
+from fastapi.exceptions import HTTPException
+
+from odoo import _, api, models
+
+from ..schema.schema import AddressInput, AddressSearch
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    @api.model
+    def _create_vals_shopinvader_address(self, data: AddressInput) -> dict:
+        state_id = self.env["res.country.state"].search(
+            [
+                "|",
+                ("name", "=", data.state),
+                ("code", "=", data.state),
+            ],
+            limit=1,
+        )
+
+        country_id = self.env["res.country"].search(
+            [
+                "|",
+                ("name", "=", data.country),
+                ("code", "=", data.country),
+            ]
+        )
+
+        vals = {
+            "name": data.name,
+            "street": data.street,
+            "street2": data.street2 or "",
+            "zip": data.zip,
+            "city": data.city,
+            "phone": data.phone or "",
+            "email": data.email or "",
+            "state_id": state_id.id,
+            "country_id": country_id.id,
+            "parent_id": self.env.context.get("authenticated_partner_id"),
```

Comment:
This value must be pass to the method by the fastapi layer. It's not a good practice to rely on values into the context...